From e09e47d9d751a8feb0555b74f1341e25f76e756c Mon Sep 17 00:00:00 2001
From: Huang Shan <huang.shan@freescale.com>
Date: Fri, 23 May 2014 14:00:18 +0800
Subject: [PATCH 68/70] ENGR00314407 swtich DHCP and static IP sometiomes cant
 be set

The device name is null, the IP settings cant be updated.
Fill the device name "eth0".

Signed-off-by: Huang Shan <huang.shan@freescale.com>
---
 Ethernet/com/fsl/ethernet/EthernetAdvDialog.java    | 15 ++++++++-------
 Ethernet/com/fsl/ethernet/EthernetConfigDialog.java |  3 +++
 Ethernet/com/fsl/ethernet/EthernetManager.java      |  4 +++-
 3 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
index b4d1323..e97a944 100644
--- a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
+++ b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
@@ -77,13 +77,14 @@ public class EthernetAdvDialog extends AlertDialog implements DialogInterface.On
 
     public void handle_saveconf() {
         EthernetDevInfo info = new EthernetDevInfo();
-	info.setIfName("eth0");
-	info.setConnectMode(mEthEnabler.getManager().getSharedPreMode());
-	info.setIpAddress(mEthEnabler.getManager().getSharedPreIpAddress());
-	info.setDnsAddr(mEthEnabler.getManager().getSharedPreDnsAddress());
-	info.setProxyAddr(mProxyIp.getText().toString());
-	info.setProxyPort(mProxyPort.getText().toString());
-	info.setProxyExclusionList(mProxyExclusionList.getText().toString());
+        String [] DevName = mEthEnabler.getManager().getDeviceNameList();
+        info.setIfName(DevName[0]);
+        info.setConnectMode(mEthEnabler.getManager().getSharedPreMode());
+        info.setIpAddress(mEthEnabler.getManager().getSharedPreIpAddress());
+        info.setDnsAddr(mEthEnabler.getManager().getSharedPreDnsAddress());
+        info.setProxyAddr(mProxyIp.getText().toString());
+        info.setProxyPort(mProxyPort.getText().toString());
+        info.setProxyExclusionList(mProxyExclusionList.getText().toString());
         mEthEnabler.getManager().updateDevInfo(info);
         mEthEnabler.getManager().setProxy();
     }
diff --git a/Ethernet/com/fsl/ethernet/EthernetConfigDialog.java b/Ethernet/com/fsl/ethernet/EthernetConfigDialog.java
index 37becb6..161724c 100644
--- a/Ethernet/com/fsl/ethernet/EthernetConfigDialog.java
+++ b/Ethernet/com/fsl/ethernet/EthernetConfigDialog.java
@@ -132,6 +132,9 @@ public class EthernetConfigDialog extends AlertDialog implements
             info.setDnsAddr(null);
         }
 
+        info.setProxyAddr(mEthEnabler.getManager().getSharedPreProxyAddress());
+        info.setProxyPort(mEthEnabler.getManager().getSharedPreProxyPort());
+        info.setProxyExclusionList(mEthEnabler.getManager().getSharedPreProxyExclusionList());
         mEthEnabler.getManager().updateDevInfo(info);
         mEthEnabler.setEthEnabled();
     }
diff --git a/Ethernet/com/fsl/ethernet/EthernetManager.java b/Ethernet/com/fsl/ethernet/EthernetManager.java
index d21ba9a..59b8d06 100644
--- a/Ethernet/com/fsl/ethernet/EthernetManager.java
+++ b/Ethernet/com/fsl/ethernet/EthernetManager.java
@@ -184,6 +184,7 @@ public class EthernetManager {
 
             try {
                 mNMService.setInterfaceDown(info.getIfName());
+                mNMService.clearInterfaceAddresses(info.getIfName());
                 mNMService.setInterfaceUp(info.getIfName());
             } catch (RemoteException re){
                 Log.e(TAG,"DHCP configuration failed: " + re);
@@ -216,7 +217,8 @@ public class EthernetManager {
 
     public EthernetDevInfo getDhcpInfo() {
         EthernetDevInfo infotemp = new EthernetDevInfo();
-        infotemp.setIfName(mTracker.getLinkProperties().getInterfaceName());
+        String [] DevName = getDeviceNameList();
+        infotemp.setIfName(DevName[0]);
         infotemp.setConnectMode(EthernetDevInfo.ETHERNET_CONN_MODE_DHCP);
         String ip;
         ip = mConnMgr.getLinkProperties(ConnectivityManager.TYPE_ETHERNET).getAddresses().toString();
-- 
1.8.0

